csrc/dynamic_transform.cpp

```diff
@@ -544,16 +531,13 @@ bool DynamicTransformConcretizer::propagateFromProducerToConsumer(
         def->toString());
 
     TORCH_INTERNAL_ASSERT(
-        id_type != IterType::Symbolic,
+        id_type.value() != IterType::Symbolic,
         "Failed to concretize ",
         root_id->toString(),
         " of ",
         consumer->toString());
 
-    auto concretized_id =
-        IterDomainBuilder(root_id).iter_type(*id_type).build();
-
-    registerMutation(root_id, concretized_id);
+    root_id->setIterType(id_type.value());
```

Comment:
@naoyam while thinking about #340 it occurred to me that we do not need to use mutator machinery if we are merely setting IterType. In the case of dynamic reshape, it was important to create new IterDomains with the proper transforms. If those transforms don't change though, we can simply use transformations like this one (and the two above it) which do not need to be registered with `registerMutation` any longer.

Reply:
I don't remember why we allow modification of IterType, but generally speaking the IR nodes are designed to be immutable since changing properties such as this could result in inconsistent fusions if done in isolation with the rest of vals. We may want to make `setIterType` `private` in the future, so I think it still makes more sense to use the mutation interface.

Reply:
Sounds good. I had the feeling there may be a reason there wasn't already a `setIterType`. I'll revert the commit and be sure the comment about iter_type option on resize is complete.

Reply:
Some properties would be fine to change, though. For example, the `BlockSync` IR node has `convertToAligned()`, which makes them to switch from `__barrier_sync(0)` to `__syncthreads()`. It should not affect anything else, so this is a pretty safe property to change.